Build for Windows: undeclared identifier 'temp_folder'

**Description**
  When I try to build sentry-native in windows, the following error occurs:
```
test_path.c:157:58: error: use of undeclared identifier 'temp_folder'
```
The fact is that this line with a comment makes the line below a comment too:
https://github.com/getsentry/sentry-native/blob/cc0889a6b1869c91239622fe233e1ca9af45ef88/tests/unit/test_path.c#L155-L156
<!-- A clear and concise description of what the bug is. -->


**When does the problem happen**

<!-- Put an `x` where relevant. -->

- [x] During build
- [ ] During run-time
- [ ] When capturing a hard crash

**Environment**

<!-- Some issues are very OS and compiler-dependent. -->

- OS: Windows 10, 64-bit
- Compiler: MinGW64 (MSYS2)
- CMake version and config: 3.17.3, SENTRY_BACKEND=crashpad, CMAKE_TOOLCHAIN_FILE=toolchains/msys2.cmake;


**Steps To Reproduce**
```
$ git clone git@github.com:getsentry/sentry-native.git
$ cd sentry-native
$ export MINGW_ROOT=/mingw64
$ pacman -S --needed - < ./toolchains/msys2-mingw64-pkglist.txt
$ cmake -GNinja -Bbuild -H. -DCMAKE_TOOLCHAIN_FILE=toolchains/msys2.cmake
$ ninja -C build
```

<!-- The best way is to provide a minimal code snippet -->

**Log output**
```
../tests/unit/test_path.c:155:33: warning: backslash and newline separated by space [-Wbackslash-newline-escape]
    // %TEMP%\\sentry_test_unit\\
                                ^
../tests/unit/test_path.c:157:58: error: use of undeclared identifier 'temp_folder'
    GetEnvironmentVariableW(L"TEMP", temp_folder, sizeof(temp_folder));
                                                         ^
../tests/unit/test_path.c:157:38: error: use of undeclared identifier 'temp_folder'
    GetEnvironmentVariableW(L"TEMP", temp_folder, sizeof(temp_folder));
                                     ^
../tests/unit/test_path.c:158:52: error: use of undeclared identifier 'temp_folder'
    sentry_path_t *path_4 = sentry__path_from_wstr(temp_folde

```
<!--
For build-related problems, paste the relevant CMake output.
For runtime problems, please use `sentry_options_set_debug(options, true)` and
paste the log output coming from sentry.
When doing so, please make sure to remove or censor your DSN from the http log output.
-->

--- a/src/sentry_sync.c
+++ b/src/sentry_sync.c
@@ -39,7 +39,7 @@ sentry__thread_setname(sentry_threadid_t thread_id, const char *thread_name)
     }
 
     // approach 2: Windows Vista+ and MSVC debugger
-#    if _WIN32_WINNT >= 0x0600
+#    if _WIN32_WINNT >= 0x0600 && defined(_MSC_VER)
     THREADNAME_INFO threadnameInfo;
     threadnameInfo.dwType = 0x1000;
     threadnameInfo.szName = thread_name;

--- a/tests/unit/test_path.c
+++ b/tests/unit/test_path.c
@@ -152,7 +152,7 @@ SENTRY_TEST(path_directory)
 #ifdef SENTRY_PLATFORM_WINDOWS
     sentry_path_t *path_3 = sentry__path_from_str("foo/bar\\baz");
 
-    // %TEMP%\\sentry_test_unit\\ 
+    // `%TEMP%\sentry_test_unit`
     wchar_t temp_folder[MAX_PATH];
     GetEnvironmentVariableW(L"TEMP", temp_folder, sizeof(temp_folder));
     sentry_path_t *path_4 = sentry__path_from_wstr(temp_folder);

